Add GitHub Button to Navbar

eyallampel1 opened the pull request:
### Description:

This PR addresses [fixes #123](https://github.com/Maakaf/maakaf-website/issues/123) where it was requested to add a GitHub button to the navbar for easier navigation to the repository. I have added a button which links directly to the repository.

[fixes #123](https://github.com/Maakaf/maakaf-website/issues/123)

#### Changes:
- Created a new component `GitHubButton` to encapsulate the GitHub logo and link logic.
- Added the `GitHubButton` component to the `Sidebar` component.
- Updated CSS to ensure proper styling and responsiveness of the new button.
  
#### Screenshots:

Before | After
-------|------
<img src="https://github.com/eyallampel1/assets/raw/main/before.png" width="300"> | <img src="https://github.com/eyallampel1/assets/raw/main/after.png" width="300">

#### Testing:
- Manually verified that the button appears as expected and links to GitHub.
- Checked the responsiveness of the navbar with the new button across different screen sizes.

#### Notes:
- Please ensure to check the styling and let me know if any changes are required.

#### Reviewers:
- Requesting reviews from @origranot @RoyTennenbaum

RoyTennenbaum commented:
Hi, thanks for taking your time to contribute @eyallampel1 :)

About the button I am not absolutely certain we'll need it there but it will be checked with @adirkandel and Efi the designer.
Other than that, all styling related issues and additions should be discussed with Efi on the discord channel.

eyallampel1 commented:
Hi @RoyTennenbaum,

Thank you for reviewing my pull request and providing feedback. I understand the importance of maintaining a consistent design and ensuring that any new additions align with the project's goals.

I am more than happy to discuss the design and functionality of the GitHub button with Efi and the rest of the team on the Discord channel. I believe that having a direct link to the GitHub repository could enhance the accessibility and openness of the project, making it easier for others to contribute. However, I am open to any modifications or even a different approach if that's what the team prefers.

I will reach out to Efi on the Discord channel to discuss the design further. In the meantime,
If there are any other changes or steps you'd recommend I take to move this PR forward, I'd be glad to hear them.

Thank you once again for your time and feedback!

Drorka commented:
> Hi, thanks for taking your time to contribute @eyallampel1 :)
> 
> About the button I am not absolutely certain we'll need it there but it will be checked with @adirkandel and Efi the designer. Other than that, all styling related issues and additions should be discussed with Efi on the discord channel.

@RoyTennenbaum note that the issue (#123) wasn't created by @eyallampel1. And in the issue it says to add it to the navbar.
I think it's worth explaining these things to the issue creator as well.

I actually agree that we should have a link to the github repo, but I think a more appropriate place would be in the footer, next to the other social links.
We can discuss this on discord with Efi.
![image](https://github.com/Maakaf/maakaf-website/assets/116891360/de453577-41ae-4305-9edb-3ed4995bbb69)

eyallampel1 commented:
Hi @RoyTennenbaum, @Drorka, and team,

Thank you all for your insightful feedback. I've taken your suggestions into consideration and made adjustments accordingly. I've moved the GitHub link to the footer alongside other social links, as @Drorka suggested, which I believe now aligns better with the project's design guidelines.

#### Updated Changes:
- Relocated the `GitHubButton` component to the footer.
- Ensured the styling conforms to the existing design.
- Updated the CSS to ensure the button is responsive and visually consistent with other social links.

#### Updated Screenshots:
Here's how the updated design looks:
Before | After
-------|------
<img src="https://github.com/eyallampel1/assets/blob/main/before_github_btn.png?raw=true" width="300"> | <img src="https://github.com/eyallampel1/assets/blob/main/after_github_btn.png?raw=true" width="300">


Additionally, I've created a gif to showcase the updated functionality:
![Updated Functionality](https://media.giphy.com/media/5UyS8uP0kIBX6eYzxQ/giphy.gif)

I believe this updated design follows more closely to the team's expectations and the existing layout of the website. I am open to any further adjustments and would be happy to continue collaborating on this to ensure it meets the project's needs.

I'm looking forward to your feedback and appreciate the opportunity to contribute to this project. Feel free to provide any additional comments or suggestions either here or on the Discord channel, and I'll be prompt in making any required changes.

Thank you once again for your time and consideration!

RoyTennenbaum commented on components/Footer/GithubIconLink.tsx:
```diff
@@ -0,0 +1,34 @@
+        >
+            <g id="GitHub-Icon">
+                <path
+                    d="M26,2C14.9543,2,6,10.9543,6,22C6,31.3894,11.7984,39.1736,20.3333,41.8C21.3333,41.9333,21.6667,41.4667,21.6667,41.0667C21.6667,40.8,21.6667,39.6,21.6667,38.0667C14.5333,39.4667,13.4667,36.1333,13.4667,36.1333C12.6,34.0667,11.4,33.6667,11.4,33.6667C10,32.8667,11.8667,32.8667,11.8667,32.8667C13.6667,32.9333,14.6667,34.6,14.6667,34.6C16.4,37.8,19.4667,37.0667,20.6667,36.6C20.86 ...
```
I am unsure about merging this one since the formatting here is off and it somehow passes our formatting checks.

Absolutely not your fault (staging should auto-format your code) and I've been looking at it in the past for a long while, I do not want to merge until figuring out what is wrong.

RoyTennenbaum reviewed (commented):
Left a comment in the code @eyallampel1

Again, formmatting is not your fault and requires checking, your PR is very nicely done and is consistent with our design :)

RoyTennenbaum commented:
fixes #123

RoyTennenbaum reviewed (approved):
Congrats on your first PR!
live-reviewed
